openstack/clustering/v1/clusters/requests.go

```diff
@@ -0,0 +1,48 @@
+package clusters
+
+import (
+	"net/http"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder Builder.
+type CreateOptsBuilder interface {
+	ToClusterCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts params
+type CreateOpts struct {
+	Name            string                 `json:"name" required:"true"`
+	DesiredCapacity int                    `json:"desired_capacity" required:"false"`
+	ProfileID       string                 `json:"profile_id" required:"true"`
+	MinSize         int                    `json:"min_size,omitempty"`
+	Timeout         int                    `json:"timeout,omitempty"`
+	MaxSize         int                    `json:"max_size,omitempty"`
+	Metadata        map[string]interface{} `json:"metadata,omitempty"`
```

Comment:
`Metadata` in other services is normally `map[string]string`. If you were to have something like:

```
metadata["foo"] = 12
metadata["bar"] = map[string]string{"some": "value"}
```

Senlin will respect the types?

Reply:
The doc specified ```A collection of key-value pairs associated with an object.``` Senlin allows bool, string, float values for metadata values. Added unit/accep tests to test metadata